### PR TITLE
added datatypes for dropna subset as per pandas-dev#434

### LINKED ISSUE #434: BUG: incorrect type hint for subset in dropna
**Describe the bug**

`dropna()` should accept any array like object.

**To Reproduce**
`pylance` in vscode
```python
import pandas as pd
data = {'col1': [1, 3, 4], 'col2': [2, 3, 5], 'col2': [2, 4, 4]}
df = pd.DataFrame(data)

df.dropna(subset=df.columns.drop('col1'))
```

**Please complete the following information:**
 - OS: [MacOS]
 - OS Version [e.g. 13]
 - python version 3.9.12
 - version of type checker ? 
 - version of installed `pandas-stubs` ?

**Additional context**
OP: https://github.com/pandas-dev/pandas/issues/49702

### PR DESCRIPTION
**Pull request by shubhankarunhale**
- [x] Closes #434  
- [x] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

File: tests/test_frame.py [hunk 1]
```diff
@@ -2332,3 +2332,14 @@ def test_getsetitem_multiindex() -> None:
     multi_index: tuple[str, str] = ("Year 1", "Q1")
     budget.loc["project A", multi_index] = 4700
     check(assert_type(budget.loc["project A", multi_index], Scalar), int)
+
+def test_frame_dropna_subset() -> None:
```
Comment (Dr-Irv) [inline tests/test_frame.py:2336]: Can you add a comment here referring to the Github issue, i.e., `# GH 434`

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
You need to run `poetry run poe style` on your code to properly format it.

File: tests/test_frame.py (hunk 1 above)
Comment (shubhankarunhale) [reply tests/test_frame.py:2337]: added comment

**Comment by shubhankarunhale** [comment]
> poetry run poe style
@Dr-Irv Formatted the test_frame.py file

**Review by Dr-Irv — APPROVED** [review]
Thanks @shubhankarunhale

**Pull request merged**